Filesystem Tier 2 support is broken when started using docker container

**Problem description**
When starting the segment store container with Tier 2 storage configured to use the filesystem (`TIER2_STORAGE=FS`), the segment store fails to start with the exception:

```
Exception in thread "main" java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: No enum constant io.pravega.segmentstore.server.store.ServiceConfig.StorageTypes.FS
    at io.pravega.segmentstore.server.host.ServiceStarter.lambda$attachStorage$2(ServiceStarter.java:181)
    at io.pravega.segmentstore.server.store.ServiceBuilder.getSingleton(ServiceBuilder.java:279)
    at io.pravega.segmentstore.server.store.ServiceBuilder.createSegmentContainerFactory(ServiceBuilder.java:259)
    at io.pravega.segmentstore.server.store.ServiceBuilder.getSingleton(ServiceBuilder.java:287)
    at io.pravega.segmentstore.server.store.ServiceBuilder.createSegmentContainerRegistry(ServiceBuilder.java:267)
    at io.pravega.segmentstore.server.store.ServiceBuilder.getSingleton(ServiceBuilder.java:287)
    at io.pravega.segmentstore.server.store.ServiceBuilder.getSegmentContainerRegistry(ServiceBuilder.java:239)
    at io.pravega.segmentstore.server.store.ServiceBuilder.access$200(ServiceBuilder.java:51)
    at io.pravega.segmentstore.server.store.ServiceBuilder$ComponentSetup.getContainerRegistry(ServiceBuilder.java:381)
    at io.pravega.segmentstore.server.host.ServiceStarter.lambda$attachZKSegmentManager$3(ServiceStarter.java:188)
    at io.pravega.segmentstore.server.store.ServiceBuilder.getSingleton(ServiceBuilder.java:279)
    at io.pravega.segmentstore.server.store.ServiceBuilder.initialize(ServiceBuilder.java:232)
    at io.pravega.segmentstore.server.host.ServiceStarter.start(ServiceStarter.java:103)
    at io.pravega.segmentstore.server.host.ServiceStarter.main(ServiceStarter.java:230)
Caused by: java.lang.IllegalArgumentException: No enum constant io.pravega.segmentstore.server.store.ServiceConfig.StorageTypes.FS
    at java.lang.Enum.valueOf(Enum.java:238)
    at io.pravega.segmentstore.server.store.ServiceConfig$StorageTypes.valueOf(ServiceConfig.java:44)
    at io.pravega.segmentstore.server.host.ServiceStarter.lambda$attachStorage$2(ServiceStarter.java:163)
    ... 13 more
```

The entrypoint script requires the value to be set as `FS` but the ServiceStarter is attempting to convert the string to an enumeration that expects `FILESYSTEM`.  This was changed in this commit: https://github.com/pravega/pravega/commit/3a1f0d57da86aa73ba0e3eeba9512c3c0a2bf885

**Problem location**
`docker/pravega/entrypoint.sh` or `segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/ServiceStarter.java`

**Suggestions for an improvement**
Either the entrypoint needs to be updated to use `FILESYSTEM` (and likely the docker-compose files would need to be updated as well), or the `ServiceConfig.StorageTypes` enumeration needs to provide a method to convert from a string that accepts `FS`.  The ServiceStarter is using `valueOf`:
```
                ServiceConfig.StorageTypes storageChoice = ServiceConfig.StorageTypes.valueOf(this.serviceConfig
                        .getStorageImplementation());
```

The StorageTypes enumeration could be updated to include a `fromString` method:

```
    public enum StorageTypes {
        HDFS("HDFS"),
        FILESYSTEM("FS"),
        INMEMORY("INMEMORY");

        private final String type;
        private StorageTypes(final String type) {
            this.type = type;
        }

        public static StorageTypes fromString(String type) {
            for (StorageTypes t : values()) {
                if (t.type.equals(type)) {
                    return t;
                }
            }
            // Fall back on valueOf
            return valueOf(type);
        }
    }
```

then ServiceStarter could be updated to use that new method:
```
                ServiceConfig.StorageTypes storageChoice = ServiceConfig.StorageTypes.fromString(this.serviceConfig
                        .getStorageImplementation());
```

--- a/docker/compose/docker-compose-nfs-mount.yml
+++ b/docker/compose/docker-compose-nfs-mount.yml
@@ -68,7 +68,7 @@ services:
     privileged: true
     environment:
       WAIT_FOR: bookie1:3181,bookie2:3182,bookie3:3183
-      TIER2_STORAGE: "FS"
+      TIER2_STORAGE: "FILESYSTEM"
       MOUNT_IN_CONTAINER: "true"
       NFS_SERVER: ${NFS_IP}:${NFS_PATH}
       NFS_MOUNT: /nfs

--- a/docker/compose/docker-compose-nfs.yml
+++ b/docker/compose/docker-compose-nfs.yml
@@ -76,7 +76,7 @@ services:
       - nfs:/nfs/
     environment:
       WAIT_FOR: bookie1:3181,bookie2:3182,bookie3:3183
-      TIER2_STORAGE: "FS"
+      TIER2_STORAGE: "FILESYSTEM"
       NFS_MOUNT: /nfs
       HDFS_REPLICATION: 1
       HDFS_URL: ${HOST_IP}:8020

--- a/docker/pravega/entrypoint.sh
+++ b/docker/pravega/entrypoint.sh
@@ -30,7 +30,7 @@ configure_tier2() {
     add_system_property "pravegaservice.storageImplementation" "${TIER2_STORAGE}"
 
     echo "Checking whether NFS mounting is required"
-    if [ "${TIER2_STORAGE}" = "FS" ] && [ "${MOUNT_IN_CONTAINER}"  = "true" ]; then
+    if [ "${TIER2_STORAGE}" = "FILESYSTEM" ] && [ "${MOUNT_IN_CONTAINER}"  = "true" ]; then
         while [ -z ${NFS_SERVER} ]
         do
             echo "NFS_SERVER not set. Looping till the container is restarted with NFS_SERVER set."

--- a/segmentstore/server/src/main/java/io/pravega/segmentstore/server/store/ServiceConfig.java
+++ b/segmentstore/server/src/main/java/io/pravega/segmentstore/server/store/ServiceConfig.java
@@ -43,7 +43,7 @@ public class ServiceConfig {
     //endregion
     public enum StorageTypes {
         HDFS("HDFS"),
-        FILESYSTEM("FS"),
+        FILESYSTEM("FILESYSTEM"),
         INMEMORY("INMEMORY");
 
         private final String type;

